python/ray/data/_internal/dataset_iterator/pipelined_dataset_iterator.py

```diff
@@ -67,16 +67,14 @@ def iter_batches(
         # Deprecated.
         prefetch_blocks: int = 0,
     ) -> Iterator[DataBatch]:
-        # Set prefetch_batches to default of 0 for DatasetPipeline.
-        return super().iter_batches(
-            prefetch_batches=prefetch_batches,
+        yield from self._base_dataset_pipeline.iter_batches(
```

Comment:
isn't this changing the behavior of `PipelineDatasetIterator`? Not using `self._base_dataset_pipeline` was intentional so that each call to `iter` would iterate over the next epoch.